internal/armada/server/event.go

```diff
@@ -49,16 +53,86 @@ func (s *EventServer) Report(ctx context.Context, message *api.EventMessage) (*t
 	if err := checkPermission(s.permissions, ctx, permissions.ExecuteJobs); err != nil {
 		return nil, status.Errorf(codes.PermissionDenied, "[Report] error: %s", err)
 	}
+
 	return &types.Empty{}, s.eventStore.ReportEvents([]*api.EventMessage{message})
 }
 
 func (s *EventServer) ReportMultiple(ctx context.Context, message *api.EventList) (*types.Empty, error) {
 	if err := checkPermission(s.permissions, ctx, permissions.ExecuteJobs); err != nil {
 		return nil, status.Errorf(codes.PermissionDenied, "[ReportMultiple] error: %s", err)
 	}
+
+	if err := s.checkForPreemptedEvents(message); err != nil {
+		return &types.Empty{}, err
+	}
+
 	return &types.Empty{}, s.eventStore.ReportEvents(message.Events)
 }
 
+func (s *EventServer) checkForPreemptedEvents(message *api.EventList) error {
+	var preemptedEvents []*api.EventMessage_Preempted
+	var jobIds []string
+
+	for _, event := range message.Events {
+		if event, ok := event.Events.(*api.EventMessage_Preempted); ok {
+			preemptedEvents = append(preemptedEvents, event)
+			if event.Preempted.JobId != "" {
+				jobIds = append(jobIds, event.Preempted.JobId)
+			}
+			if event.Preempted.PreemptiveJobId != "" {
+				jobIds = append(jobIds, event.Preempted.PreemptiveJobId)
+			}
+		}
+	}
+
+	if len(preemptedEvents) == 0 {
+		return nil
+	}
+
+	results, err := s.jobRepository.GetJobsByIds(jobIds)
+	if err != nil {
+		return errors.WithMessage(err, "error fetching jobs for preempted and preemptive job ids")
+	}
+
+	for _, event := range preemptedEvents {
+		if err := s.enrichPreemptedEvent(event, results); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *EventServer) enrichPreemptedEvent(event *api.EventMessage_Preempted, results []*repository.JobResult) error {
```

Comment:
I'd suggest giving the second argument more descriptive name than `results`.  Maybe `JobDetails`.  Also as you need to do lookoups on this, would it better to make it a map[string]JobResult?